src/averageAges.js

```diff
@@ -36,8 +52,31 @@ function calculateMenAverageAge(people, century) {
  *
  * @return {number}
  */
+
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const arrWomen = people.filter(person => person.sex === 'f');
+
+  const calcAgeWomenWithChildren = function() {
+    const mother = people.map(woman => woman.mother);
+    const arrWomenWithChidren = arrWomen.filter(woman =>
+      mother.includes(woman.name));
+
+    const initialValue = 0;
+    const sumAgeWomen = arrWomenWithChidren.reduce((a, b) =>
+      a + (b.died - b.born), initialValue);
+
+    return Number((sumAgeWomen / arrWomenWithChidren.length).toFixed(2));
+  };
+
+  const calcAgeWomen = function() {
+    const initialValue = 0;
+    const sumAgeWomen = arrWomen.reduce((a, b) => a + (b.died - b.born),
+      initialValue);
+
+    return Number((sumAgeWomen / arrWomen.length).toFixed(2));
+  };
```

Comment:
you can create just one function and use it for both men and women. Specify an additional condition in another way